File: test/predict.py
import sys
import h5py
import caffe
import time
import numpy as np

# Usage: python predict.py foo.net foo.caffemodel iters y.h5
# Don't know how to feed in x here as it is hardcoded in the hdf5 layer

caffe.set_phase_test()
caffe.set_mode_gpu()
net = caffe.Net(sys.argv[1], sys.argv[2])
iters = int(sys.argv[3])
ydims = net.blobs['fc2'].data.shape
batch = ydims[0]
ydims = (iters*ydims[0], ydims[1], ydims[2], ydims[3])
y = np.zeros(ydims, dtype=net.blobs['fc2'].data.dtype)
yptr = 0
t = time.time()

for i in range(0,iters):
    net.forward()
    y[yptr:yptr+batch,:,:,:] = net.blobs['fc2'].data
    yptr = yptr+batch

print 'Elapsed: %s' % (time.time() - t)

f = h5py.File(sys.argv[4])
f.create_dataset('data', data=y.squeeze())
f.close()

File: test/predict.jl
using HDF5
using KUnet
@time x = h5read(ARGS[1], "/data")
@time l1 = KUnet.Layer(ARGS[2])
@time l2 = KUnet.Layer(ARGS[3])
net = [l1,l2]
@time gc()
@time y = KUnet.predict(net, x, batch=937)
@time y = KUnet.predict(net, x, batch=937)
# Profile.print()
@time h5write(ARGS[4], "data", y)
